### PR TITLE
feat: add site top pages data access (SITES-21274)

### PR DESCRIPTION
**Pull request by iuliag**
Implements data model/access for: https://jira.corp.adobe.com/browse/SITES-21274

### CODE REVIEW

File: packages/spacecat-shared-data-access/src/dto/site-top-page.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   * @returns {{siteId, url, traffic, source, geo, importedAt, SK: string}}
+   */
+  toDynamoItem: (siteTopPage) => ({
+    siteId: siteTopPage.getSiteId(),
+    url: siteTopPage.getURL(),
+    traffic: siteTopPage.getTraffic(),
+    source: siteTopPage.getSource(),
+    geo: siteTopPage.getGeo(),
+    importedAt: siteTopPage.getImportedAt(),
+    SK: `${siteTopPage.getSource()}#${siteTopPage.getGeo()}#${siteTopPage.getTraffic()}`,
```
Comment (solaris007) [inline packages/spacecat-shared-data-access/src/dto/site-top-page.js:28]: i guess for the traffic we need left zero padding for lexicographical sorting. should pick a "realistic" top number

**Review by solaris007 — COMMENTED** [review]
direction is looking great!

File: packages/spacecat-shared-data-access/test/it/db.test.js [hunk 2]
```diff
@@ -740,4 +754,66 @@ describe('DynamoDB Integration Test', async () => {
...
+    expect(topPage.getTraffic()).to.equal(siteTopPageData.traffic);
+    expect(topPage.getSource()).to.equal(siteTopPageData.source);
+    expect(topPage.getGeo()).to.equal(siteTopPageData.geo);
+    expect(topPage.getImportedAt()).to.equal(siteTopPageData.importedAt);
+  });
+
+  it('retrieves top pages for a site from a specific source and geo in descending traffic order', async () => {
+    const siteId = (await dataAccess.getSites())[0].getId();
+
+    const siteTopPages = await dataAccess.getTopPagesForSite(siteId, 'ahrefs', 'global');
```
Comment (iuliag) [inline packages/spacecat-shared-data-access/test/it/db.test.js:798]: @AndreiAlexandruParaschiv , here's an example of how you'd get top pages for a site for the audit.

**Review by solaris007 — APPROVED** [review]
LGTM

**Pull request merged**